src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs

```diff
@@ -9,22 +9,96 @@
 
 namespace System.Reflection
 {
+    internal unsafe struct NativeAssemblyNameParts
+    {
+        public char* _pName;
+        public ushort _major, _minor, _build, _revision;
+        public char* _pCultureName;
+        public byte* _pPublicKeyOrToken;
+        public int _cbPublicKeyOrToken;
+        public AssemblyNameFlags _flags;
+
+        //
+        // Native AssemblyName store uint16 components for the version. Version and AssemblyVersion store int32 or uint32.
+        // When the former are initialized from the latter, the components are truncated to uint16 size.
+        // When the latter are initialized from the former, they are zero-extended to int32 size. For uint16 components, the max
+        // value is used to indicate an unspecified component.
+        //
+
+        public void SetVersion(Version? version)
+        {
+            if (version != null)
+            {
+                _major = (ushort)version.Major;
+                _minor = (ushort)version.Minor;
+                _build = (ushort)version.Build;
+                _revision = (ushort)version.Revision;
+            }
+            else
+            {
+                _major = 0;
```

Comment:
Should this initialize to `ushort.MaxValue`?

Reply:
Fixed

Reply:
Actually, it depends. It should be 0 for one caller and `ushort.MaxValue` for the other caller.